crates/pallet-transaction-fees/src/lib.rs

```diff
@@ -185,11 +205,16 @@ where
             compute: BalanceOf::<T>::zero(),
             tips: BalanceOf::<T>::zero(),
         });
+
+        // Move the `NextTransactionByteFee` value into the current `TransactionByteFee`
+        TransactionByteFee::<T>::put(NextTransactionByteFee::<T>::get());
     }
 
     // TODO: Fees will be split between farmers and executors in the future
     fn do_finalize(_n: BlockNumberFor<T>) {
+        // Clear the current `TransactionByteFee` and update the value for `NextTransactionByteFee`
         TransactionByteFee::<T>::take();
```

Comment:
What do you think about not clearing? It seems useful to be able to know what the fee was in a particular block using runtime API after the block was queried or using storage query via RPC.

If we do that, then I'd probably put both current and next value in the same storage item for efficiency.

Reply:
>`fn transaction_byte_fee()` will return the value of `TransactionByteFee` for executing the extrinsic of the current block, and return the value of `NextTransactionByteFee` for validating extrinsic to be included in the next block

If we do not clear `TransactionByteFee` we are unable to do the above unless we introduce another runtime API `next_transaction_byte_fee`

Reply:
Above what? I do not understand.

Reply:
>fn transaction_byte_fee() will return the value of TransactionByteFee for executing the extrinsic of the current block, and return the value of NextTransactionByteFee for validating extrinsic to be included in the next block